lib/core-integration/src/Test/Integration/Framework/TestData.hs

```diff
@@ -621,6 +622,9 @@ errMsg400ScriptNotUniformRoles :: String
 errMsg400ScriptNotUniformRoles =
     "All keys of a script must have the same role: either payment or delegation."
 
+errMsg403OutputsAlreadyCovered :: String
+errMsg403OutputsAlreadyCovered =
+    "The transaction to be balanced has already all outputs covered. Please send a transaction that requires more inputs to be picked to be balanced."
```

Comment:
yes, message is unfortunate for two reasons: first of all, unbalanced tx is the one that lacks outputs OR inputs, Secondly, we should rather write that sent tx is balanced and does not need further balancing but signing and then submitting.

Reply:
I resigned from this error naming in https://github.com/input-output-hk/cardano-wallet/pull/2900

Reply:
I was more wondering: why not remove the error?

If the endpoint succeeds in the — I take rare event of PAB submitting an already balanced tx, then they wouldn't have to add special handling for:
> we should rather write that sent tx is balanced and does not need further balancing but signing and then submitting.

Reply:
Also seemed to me that it _might_ be easier for _us too_, to not care about defining `areOutputsCovered`. I would presume the code responsible for balancing the tx would notice that the tx was already balanced and simply return. 

But not sure how the underlying balancing pieces fit together (glanced at it a bit but didn't grasp it yet)

Reply:
The problem is that I am reusing ApiCoinSelection in the response of endpoint and it cannot be empty.So in case of already balance tx I tentatively choose to use some error to circumvent this problem. As a first step.